packages/stylelint-plugin/README.md

```diff
@@ -0,0 +1,77 @@
+# Kaizen Stylelint Plugin
+
+## Tl;dr
+
+`yarn add @kaizen/stylelint-plugin`
+
+Then, add the plugin and the rules you want to your stylelint configuration (e.g. `stylelint.config.js` or `.stylelintrc.json`):
+
+```json
+{
+  "plugins": ["@kaizen/stylelint-plugin"],
+  "rules": {
+    "kaizen/no-deprecated-tokens": [
+      true,
+      { "severity": "warning", "allowFixing": true }
+    ],
+    "kaizen/no-transitive-tokens": [
+      true,
+      { "severity" : "warning", "allowFixing": true }
+    ],
+    "kaizen/no-invalid-functions": [
+      true,
+      { "severity": "warning", "allowFixing": true }
+    ],
+    "kaizen/no-invalid-equations": [
+      true,
+      { "severity": "warning", "allowFixing": true }
+    ],
+    "kaizen/imports-no-unused": [
+      true,
+      { "severity": "warning", "allowFixing": true }
+    ],
+    "kaizen/imports-no-extraneous": [
+      true,
+      { "severity": "warning", "allowFixing": true }
+    ]
+  }
+}
+
+```
+
+## Rules
+- `kaizen/no-deprecated-tokens` - Detects usages of deprecated tokens, with the ability to automatically migrate many instances.
+- `kaizen/no-invalid-functions` - Detects invalid usages of Kaizen tokens within some functions that are known to be problematic such as `rgba`, `darken` or `mix`, and automatically fixes `rgba`, `rgb`, `add-alpha` instances by ensuring the correct `-rgb-params` tokens are used, OR by replacing unfixable color manipulation functions into their computed values (see [Sass Manipulate Color Functions] (https://www.w3schools.com/sass/sass_functions_color.asp))
+- `kaizen/no-invalid-equations` - Detects invalid usages of Kaizen tokens within equations. It won't be able to automatically fix many cases, but it can for some simple cases for example negating a Kaizen variable without a `calc()` function, which is going to be unsupported in the future due to CSS variables.
+- `kaizen/no-transitive-tokens` - Prevent and fix usages of local non-Kaizen variables (only ones that are defined on the current sheet, ignoring any that are imported) that contain Kaizen tokens in their values. For example, `$card-bg: $kz-color-wisteria-800; .card { background-color: $card-bg; }` would be disallowed, and would be autofixed to `background-color: $kz-color-wisteria-800;`
```

Comment:
How desirable is this behaviour? I can see people using local variables as "decision tokens".

In the example you have setting `$card-bg: $kz-color-wisteria-800` makes it really clear that's a design decision, and if the `$card-bg` decision token is used in multiple places it makes it easy to update to the decision on what card background colours are (compared to a find/replace of the wisteria-800 option token, which might be used for things other than card backgrounds too)

Reply:
(Apologies if this was previously discussed)

Reply:
It's absolutely a tricky one.
It massively helped in the migration of murmur though, because doing those replacements allowed the other rules to run.
Apart from murmur, i haven't been using it. It might not be that useful outside of migrations.